src/plugins/links/public/types.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+
+import {
+  HasEditCapabilities,
+  HasInPlaceLibraryTransforms,
+  HasParentApi,
+  HasType,
+  PublishesPanelDescription,
+  PublishesPanelTitle,
+  PublishesSavedObjectId,
+  PublishesUnifiedSearch,
+  SerializedTitles,
+} from '@kbn/presentation-publishing';
+import { DefaultEmbeddableApi } from '@kbn/embeddable-plugin/public';
+import { DynamicActionsSerializedState } from '@kbn/embeddable-enhanced-plugin/public/plugin';
+import { HasSerializedChildState, PresentationContainer } from '@kbn/presentation-containers';
+import { LocatorPublic } from '@kbn/share-plugin/common';
+import { DashboardLocatorParams, DASHBOARD_CONTAINER_TYPE } from '@kbn/dashboard-plugin/public';
+import { DashboardAttributes } from '@kbn/dashboard-plugin/common';
+
+import { CONTENT_ID } from '../common';
+import { Link, LinksAttributes, LinksLayoutType } from '../common/content_management';
+
+export type LinksParentApi = PresentationContainer &
+  HasType<typeof DASHBOARD_CONTAINER_TYPE> &
+  HasSerializedChildState<LinksSerializedState> &
+  PublishesSavedObjectId &
+  PublishesPanelTitle &
+  PublishesPanelDescription &
+  PublishesUnifiedSearch & {
+    locator?: Pick<LocatorPublic<DashboardLocatorParams>, 'navigate' | 'getRedirectUrl'>;
+  };
+
+export type LinksApi = HasType<typeof CONTENT_ID> &
+  DefaultEmbeddableApi<LinksSerializedState, LinksRuntimeState> &
+  HasParentApi<LinksParentApi> &
+  HasEditCapabilities &
+  HasInPlaceLibraryTransforms<LinksRuntimeState> & {
+    layout$: BehaviorSubject<LinksLayoutType | undefined>;
+    links$: BehaviorSubject<ResolvedLink[] | undefined>;
```

Comment:
Should these be part of the public API?

Reply:
Good call. Removed in https://github.com/elastic/kibana/pull/178670/commits/3084f31eca0552b70b44716d6663e17998bd6926